whole-tale/girder_wholetale
# Convert update_citations to async event

### Motivation
Whenever user modifies external data (Tale View > Files > Data) there's a long gap between hitting a submit button and moment in time when dashboard shows dataset change in the file browser. It's caused mostly by the fact we're synchronously generating datasetCitation info and related_identifiers field on Tale.

### Solution
This PR converts `update_citation` to an asynchronous event.

### How to test?
1. Deploy, create a Tale, register a few datasets (e.g.: doi:10.5281/zenodo.3381658, doi:10.5065/D6862DM8, doi:10.5063/F10K26VR)
2. Using `Tale View > Files > Data` create a dataSet using random combination of files from different datasets.
3. Inspect Tale object (or use dashboards event PR) to verify that both datasetCitation and relatedIdentifiers change accordingly.
